ownership-chain/runtime/src/migrations/v1.rs

```diff
@@ -0,0 +1,90 @@
+/// Migration v1 for the runtime.
+///
+/// This migration is responsible for migrating accounts from AccountId32 to AccountId20.
+/// It only migrates [`pallet_sudo::Key`].
+
+/// Unchecked version migration logic.
+pub mod version_unchecked {
+	use frame_support::{
+		storage::{storage_prefix, unhashed},
+		traits::{Get, OnRuntimeUpgrade, PalletInfoAccess},
+	};
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	/// Migrate Sudo key
+	pub struct MigrateSudo<T>(sp_std::marker::PhantomData<T>);
+
+	impl<T> OnRuntimeUpgrade for MigrateSudo<T>
+	where
+		T: pallet_sudo::Config,
+	{
+		/// This simply asserts that it's not possible to read the old sudo key, since it is
+		/// `AccountId32`.
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			// make sure we can't read current sudo key
+			let raw_sudo_key = storage_prefix(
+				<pallet_sudo::Pallet<T> as PalletInfoAccess>::name().as_bytes(),
+				b"Key",
+			);
+			let maybe_sudo =
+				unhashed::get_raw(raw_sudo_key.as_ref()).expect("Sudo key should be set!; qed");
+
+			frame_support::log::debug!("Old sudo key: {:?}", maybe_sudo);
+
+			// old sudo key has length of 32
+			assert_eq!(maybe_sudo.len(), 32);
+
+			Ok(Vec::new())
+		}
+
+		/// Migrate sudo key.
+		///
+		/// Simply remove old sudo key and insert new sudo key.
+		///
+		/// This function is called during the runtime upgrade process.
+		///
+		/// NOTE: storage version of `pallet_sudo` can not be updated because it's an external
+		/// pallet, i.e we have to set `#[pallet::storage_version(VERSION)]` attribute.
```

Comment:
how do we ensure that migration only runs once then?

Reply:
we remove it once it's run. i.e next runtime upgrade we don't run anything from `v1.rs`